lib/bootstrap.rb

```diff
@@ -0,0 +1,99 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+# This file is designed to bootstrap a `Bundler.require` based Ruby app (such as
+# a Ruby on Rails app) so that the app can be instrumented and observed by the
+# New Relic Ruby agent without the agent being added to the app as a dependency.
+#
+# Given a production-ready Ruby app that optionally has a pre-packaged "frozen"
+# or "deployment" based gem bundle, the New Relic Ruby agent can be introduced
+# to the app without modifying the app and keeping all of the app's content
+# read-only.
+#
+# Prerequisites:
+#   - Ruby (tested v2.4+)
+#   - Bundler (included with Ruby, tested v1.17+)
+#
+# Instructions:
+#   - First, make sure the New Relic Ruby agent exists on disk. For these
+#     instructions, we'll assume the agent exists at `/newrelic`.
+#     - The agent can be downloaded as the "newrelic_rpm" gem from RubyGems.org
+#       and unpacked with "gem unpack"
+#     - The agent can be cloned from the New Relic public GitHub repo:
+#       https://github.com/newrelic/newrelic-ruby-agent
+#   - Next, use the "RUBYOPT" environment variable to require ("-r") this
+#     file (note that the ".rb" extension is dropped):
+#       ```
+#       export RUBYOPT="-r /newrelic/lib/bootstrap"
+#       ```
+#   - Launch an existing Ruby app as usual. For a Ruby on Rails app, this might
+#     involve running `bin/rails server`.
+#   - In the Ruby app's directory, look for and inspect
+#     `log/newrelic_agent.log`. If this file exists and there are no "WARN" or
+#     "ERROR" entries within it, then the agent was successfully introduced to
+#     the Ruby application.
```

Comment:
Love these instructions. Does someone also need to pass a license key? If so, should that be called out here?

Reply:
At some point a license key and some required configs are needed... but I'm not sure we want to encourage independent use of introducing the agent in this way quite yet. We actually talked about gating this behind an env var. 

But on the other hand, these instructions get you 90% of the way towards a working agent, so might as well include the final steps. 

I think I'd like to add both a disclaimer: `NOTE: introducing the agent into your application via bootstrap is in beta.`, as well as the final steps for a working agent (license key and config file). 

What do you think?

Reply:
that sounds great! on the beta line, maybe append something like "use at your own risk"

i'm not sure if you necessarily need a config file to make the agent work since almost everything can be set by env vars. i think it will just fall back to defaults, but i'm not certain.

Reply:
Oh you're totally right. You'll still get a `WARN` about a missing config, but things seem to work

Reply:
https://github.com/newrelic/newrelic-ruby-agent/pull/2635/commits/2d05f8e381aab52bec1e3f94f13355ab95efd481